src/test/java/org/stellar/sdk/AuthTest.java

```diff
@@ -0,0 +1,606 @@
+                            .contractAddress(new Address(contractId).toSCAddress())
+                            .functionName(Scv.toSymbol("increment").getSym())
+                            .args(new SCVal[0])
+                            .build())
+                    .build())
+            .subInvocations(new SorobanAuthorizedInvocation[0])
+            .build();
+    SorobanAuthorizationEntry entry =
+        new SorobanAuthorizationEntry.Builder()
+            .credentials(credentials)
+            .rootInvocation(invocation)
+            .build();
+
+    SorobanAuthorizationEntry signedEntry =
+        Auth.authorizeEntry(entry, signer, validUntilLedgerSeq, network);
+
+    HashIDPreimage preimage =
+        new HashIDPreimage.Builder()
+            .discriminant(EnvelopeType.ENVELOPE_TYPE_SOROBAN_AUTHORIZATION)
+            .sorobanAuthorization(
+                new HashIDPreimage.HashIDPreimageSorobanAuthorization.Builder()
+                    .networkID(new Hash(network.getNetworkId()))
+                    .nonce(new Int64(123456789L))
+                    .invocation(invocation)
+                    .signatureExpirationLedger(
+                        new Uint32(new XdrUnsignedInteger(validUntilLedgerSeq)))
+                    .build())
+            .build();
+    byte[] signature = signer.sign(Util.hash(preimage.toXdrByteArray()));
+    SorobanCredentials expectedCredentials =
+        new SorobanCredentials.Builder()
+            .discriminant(SorobanCredentialsType.SOROBAN_CREDENTIALS_ADDRESS)
+            .address(
+                new SorobanAddressCredentials.Builder()
+                    .address(new Address(signer.getAccountId()).toSCAddress())
+                    .nonce(new Int64(123456789L))
+                    .signatureExpirationLedger(
+                        new Uint32(new XdrUnsignedInteger(validUntilLedgerSeq)))
+                    .signature(
+                        Scv.toVec(
+                            Collections.singleton(
+                                Scv.toMap(
+                                    new LinkedHashMap<SCVal, SCVal>() {
+                                      {
+                                        put(
+                                            Scv.toSymbol("public_key"),
+                                            Scv.toBytes(signer.getPublicKey()));
+                                        put(Scv.toSymbol("signature"), Scv.toBytes(signature));
+                                      }
+                                    }))))
+                    .build())
+            .build();
+
+    SorobanAuthorizationEntry expectedEntry =
+        new SorobanAuthorizationEntry.Builder()
+            .credentials(expectedCredentials)
+            .rootInvocation(invocation)
+            .build();
+    assertEquals(expectedEntry, signedEntry);
+    assertNotSame(entry, signedEntry);
+  }
+
+  @Test
+  public void testSignAuthorizeEntryWithBase64EntryAndFunctionSigner() throws IOException {
```

Comment:
for consideration later, the duplicate test case variants for base64 and xdr object graph parameters could probably be reduced to just the base64 parameters, as that will by default exercise the xdr object path internally as well correct? would just. be less test code to maintain over time.

Reply:
Yes, I will remove it later.